src/bridge/cockpitdbuscache.c

```diff
@@ -544,15 +544,29 @@ introspect_next (CockpitDBusCache *self)
         }
       else
         {
-          g_debug ("%s: calling Introspect() on %s", self->logname, id->path);
-
-          id->introspecting = TRUE;
-          g_dbus_connection_call (self->connection, self->name, id->path,
-                                  "org.freedesktop.DBus.Introspectable", "Introspect",
-                                  g_variant_new ("()"), G_VARIANT_TYPE ("(s)"),
-                                  G_DBUS_CALL_FLAGS_NONE, -1,
-                                  self->cancellable, on_introspect_reply,
-                                  g_object_ref (self));
+          GDBusInterfaceInfo *iface = NULL;
+          if (id->interface)
+            iface = cockpit_dbus_interface_info_lookup (self->introspected, id->interface);
```

Comment:
Well done figuring this out! If this PR needs some more changes, then it would be worthwhile sending this commit as a separate PR.

When reading this I was wondering if `iface` is being leaked here, as it's not unreffed anywhere. But that's just a hash table lookup, so this doesn't take a ref. Would be nice if the return value was `const` to point that out. We don't actually call `cockpit_dbus_interface_info_lookup()` anywhere outside of cockpitdbuscache.c, so this could even become an internal static function and change API at will.